lib/matplotlib/cycle.py

```diff
@@ -0,0 +1,140 @@
+import itertools
+import six
+from matplotlib import rcParams
+
+
+class Cycle:
+
+    def __init__(self):
+        """
+        Set the initial cycle of styles to be used by the lines of the graph
+        """
+        self._styles = {'color': None,
+                        'linestyle': None,
+                        'linewidth': None,
+                        'marker': None,
+                        'markersize': None,
+                        'markeredgewidth': None,
+                        'markeredgecolor': None,
+                        'markerfacecolor': None,
+                        'antialiased': None,
+                        'dash_capstyle': None,
+                        'solid_capstyle': None,
+                        'dash_joinstyle': None,
+                        'solid_joinstyle': None,
+                        }
+        self._styles_list = {}
+        self.set_color_cycle()
+        self.set_line_cycle()
+
+    def __getstate__(self):
+        return {'_styles_list': self._styles_list}
+
+    def __setstate__(self, state):
+        self.__init__()
+        self.__dict__.update(state)
+        for style in self._styles_list.keys():
+            self.set_cycle(style, self._styles_list[style])
+
+    def next(self, args={}):
+        """
+        Returns the next set of line attributes for a line on the graph to use
+        *args* is an optional dictionary of style arguments
+        Styles that already exist in *args* will not be cycled through
+        """
+        args = args.copy()
```

Comment:
I don't really like this pattern.  It requires a good deal of work on the part of the user to do anything but the 'inner' product of the cycles and the ability to skip cycling one of the styles seems like it will produce a good deal of surface area for confusion.